## 1.1 - Organização e descrição da telemetria


Baseado nas informações geradas, podemos interpretar os dados da seguinte maneira:

TEMPERATURA (ºC)

  - TEMPERATURA INTERNA: - 22ºC até 48ºC.
  
    No CSV gerado por IA, a temperatura sobe continuamente indicando uma perda progressiva do controle Térmico.
  - TEMPERATURA EXTERNA:

    a temperatura externa cai cada vez mais, demonstrando um possivel afastamente de fonte de calor.
  
  O sistema térmico não esta conseguindo compensar o frio externo, resultando em um superaquecimento interno.

Integridade Estrutural (0/1)

    - Inicialmente: 1 (integro)

    - A partir de 01:03:00: 0 (Falha Estrutural)


  A estrutura se mantém estável até um certo pónto, após isso, ocorre o colapso estrutural da nave, provavelmente causado por:
    - estresse térmico,

    - queda de pressão,

    - falha em sistemas críticos.


NÍVEIS DE ENERGIA
    - Queda contínua de 92% - 42%

  o Consumo elevado sem reposição, possivelmente causado por sobrecarga dos sistemas ao tentar compensar as falhas, especialmente no sistema térmico, faz com que comece a aparecer problemas em outras áreas, até que culmine na falha total do sistema.

PRESSÃO DOS TANQUES
    - Redução constante de 5.2 - 2.9 bar
  
  Possívelmente causado por um vazamento, ou um consumo acelerado, acarretando em falhas críticas quando a pressão chega abaixo de 4.0 bar.

MODULOS CRÍTICOS
    - Inicialmente: todos OK

    - Progressão:

      - Termal: Primeiro a falhar

      - Comunicação: começa a degradar

      - Propulsão: entra em falha após

      - Vida: último a falhar

    - Estado final:
      - Todos os modulos em falha
    

## 1.2 - Algoritmo de verificação



1.   importar biblioteca pandas
2.   importar dados do CSV
3.   definir estado
4.   definir MOTIVOS

5.   se integridade estrutural == 0
        retornar - falha estrutural
6.   se temperatura interna > 35
        retornar - temperatura interna crítica
7.   se nivel de energia < 60
        retornar - energia insuficiente
8.   se pressao do tanque < 4.0
        retornar pressão dos tanques baixa
9.   se modulos =/ 1
        retornar - FALHA
10.   se motivos == 0
        retornar - LANÇAMENTO PERMITIDO
      se não - retornar - LANÇAMENTO NÃO PERMITIDO
11.  Print resultado
      se problemas  
         imprimir p in problemas



# 1.3 Script em Python

In [34]:
import pandas as pd
print('ok')

ok


In [35]:
from google.colab import files
uploaded = files.upload()
uploaded


Saving Dados_tel.csv to Dados_tel (4).csv


{'Dados_tel (4).csv': b'timestamp,temperatura_interna_c,temperatura_externa_c,integridade_estrutural,nivel_energia_pct,pressao_tanque_bar,modulo_vida,modulo_propulsao,modulo_comunicacao,modulo_termico\r\n2026-03-24T00:00:00Z,22,-55,1,92,5.2,OK,OK,OK,OK\r\n2026-03-24T00:10:00Z,23,-57,1,90,5.1,OK,OK,OK,OK\r\n2026-03-24T00:20:00Z,24,-60,1,87,5.0,OK,OK,OK,OK\r\n2026-03-24T00:30:00Z,26,-62,1,83,4.9,OK,OK,OK,OK\r\n2026-03-24T00:40:00Z,28,-65,1,78,4.8,OK,OK,OK,WARN\r\n2026-03-24T00:50:00Z,30,-67,1,74,4.7,OK,OK,OK,WARN\r\n2026-03-24T01:00:00Z,32,-70,1,70,4.5,OK,OK,OK,FAIL\r\n2026-03-24T01:10:00Z,35,-72,1,66,4.3,OK,OK,WARN,FAIL\r\n2026-03-24T01:20:00Z,38,-75,1,61,4.1,OK,WARN,WARN,FAIL\r\n2026-03-24T01:30:00Z,40,-78,0,58,3.8,WARN,WARN,FAIL,FAIL\r\n2026-03-24T01:40:00Z,42,-80,0,52,3.5,WARN,FAIL,FAIL,FAIL\r\n2026-03-24T01:50:00Z,45,-82,0,47,3.2,FAIL,FAIL,FAIL,FAIL\r\n2026-03-24T02:00:00Z,48,-85,0,42,2.9,FAIL,FAIL,FAIL,FAIL'}

In [36]:
df = pd.read_csv(list(uploaded.keys())[0])
display(df)

,timestamp,temperatura_interna_c,temperatura_externa_c,integridade_estrutural,nivel_energia_pct,pressao_tanque_bar,modulo_vida,modulo_propulsao,modulo_comunicacao,modulo_termico
0,2026-03-24T00:00:00Z,22,-55,1,92,5.2,OK,OK,OK,OK
1,2026-03-24T00:10:00Z,23,-57,1,90,5.1,OK,OK,OK,OK
2,2026-03-24T00:20:00Z,24,-60,1,87,5.0,OK,OK,OK,OK
3,2026-03-24T00:30:00Z,26,-62,1,83,4.9,OK,OK,OK,OK
4,2026-03-24T00:40:00Z,28,-65,1,78,4.8,OK,OK,OK,WARN
5,2026-03-24T00:50:00Z,30,-67,1,74,4.7,OK,OK,OK,WARN
6,2026-03-24T01:00:00Z,32,-70,1,70,4.5,OK,OK,OK,FAIL
7,2026-03-24T01:10:00Z,35,-72,1,66,4.3,OK,OK,WARN,FAIL
8,2026-03-24T01:20:00Z,38,-75,1,61,4.1,OK,WARN,WARN,FAIL
9,2026-03-24T01:30:00Z,40,-78,0,58,3.8,WARN,WARN,FAIL,FAIL


In [37]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13 entries, 0 to 12
Data columns (total 10 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   timestamp               13 non-null     object 
 1   temperatura_interna_c   13 non-null     int64  
 2   temperatura_externa_c   13 non-null     int64  
 3   integridade_estrutural  13 non-null     int64  
 4   nivel_energia_pct       13 non-null     int64  
 5   pressao_tanque_bar      13 non-null     float64
 6   modulo_vida             13 non-null     object 
 7   modulo_propulsao        13 non-null     object 
 8   modulo_comunicacao      13 non-null     object 
 9   modulo_termico          13 non-null     object 
dtypes: float64(1), int64(4), object(5)
memory usage: 1.1+ KB


In [38]:
df.describe()

,temperatura_interna_c,temperatura_externa_c,integridade_estrutural,nivel_energia_pct,pressao_tanque_bar
count,13.000000,13.000000,13.000000,13.000000,13.000000
mean,33.307692,-69.846154,0.692308,69.230769,4.307692
std,8.721474,9.805153,0.480384,16.563785,0.755493
min,22.000000,-85.000000,0.000000,42.000000,2.900000
25%,26.000000,-78.000000,0.000000,58.000000,3.800000
50%,32.000000,-70.000000,1.000000,70.000000,4.500000
75%,40.000000,-62.000000,1.000000,83.000000,4.900000
max,48.000000,-55.000000,1.000000,92.000000,5.200000


In [43]:
import pandas as pd

file_path = 'Dados_tel.csv'

data_frame = pd.read_csv(file_path)

df = pd.read_csv(file_path)

print(df)

def avaliar_lancamento(df_data):
    df_data['pressao_tanque_bar'] = pd.to_numeric(df_data['pressao_tanque_bar'], errors='coerce')

    estado = df_data.iloc[-1]

    motivos = []

    if estado["integridade_estrutural"] == 0:
        motivos.append("Falha estrutural")

    if estado["temperatura_interna_c"] > 35:
        motivos.append("Temperatura interna crítica")

    if estado["nivel_energia_pct"] < 60:
        motivos.append("Energia insuficiente")

    if estado["pressao_tanque_bar"] < 4.0:
        motivos.append("Pressão dos tanques baixa")

    modulos = [
        "modulo_vida",
        "modulo_propulsao",
        "modulo_comunicacao",
        "modulo_termico"
    ]

    for m in modulos:
        if estado[m] == "FAIL":
            motivos.append(f"Falha no {m}")

    # Resultado final
    if len(motivos) == 0:
        return "✅ LANÇAMENTO PERMITIDO", []
    else:
        return "❌ LANÇAMENTO NÃO PERMITIDO", motivos


# Exemplo de uso
status, problemas = avaliar_lancamento(df)

print(status)

if problemas:
    print("\nMotivos:")
    for p in problemas:
        print("- ", p)

               timestamp  temperatura_interna_c  temperatura_externa_c  \
0   2026-03-24T00:00:00Z                     22                    -55   
1   2026-03-24T00:10:00Z                     23                    -57   
2   2026-03-24T00:20:00Z                     24                    -60   
3   2026-03-24T00:30:00Z                     26                    -62   
4   2026-03-24T00:40:00Z                     28                    -65   
5   2026-03-24T00:50:00Z                     30                    -67   
6   2026-03-24T01:00:00Z                     32                    -70   
7   2026-03-24T01:10:00Z                     35                    -72   
8   2026-03-24T01:20:00Z                     38                    -75   
9   2026-03-24T01:30:00Z                     40                    -78   
10  2026-03-24T01:40:00Z                     42                    -80   
11  2026-03-24T01:50:00Z                     45                    -82   
12  2026-03-24T02:00:00Z              

# 1.4 Análise energética



Como os dados não foram diretamente infornmados, realizei os calculos baseados em uma estimativa baseado no comportamento do nivel de energia ao longo do tempo, junto com valores típicos para realizar a analise.



*   Capacidade Total : 100 kWh

  Energia inicial:
          
          Energia inicial = 92% x 100 = 92kWh

*   Consumo por hora:
          
          intervalo de tempo = 2 horas
          Consumo = Queta total = 92 - 42 = 50%
          consumo = 50 %
          consumo por hora = 50 / 2 = 25
          consumo por hora = 25 %
          CONVERSÃO PARA KWH

          E.Inicial = 92% x 100 = 92 kWh
          Consumo por hora = 25% x 100 = 25kWh/h
        
*   Consumo na decolagem:

          C= 25 x 1.6 = 40 kWh/h
*   Autonomia
          
          autonomia = energia / Consumo
          autonomia = 92 / 28.75
          autonomia = 3.2 horas

      Considerando cenario com decolagem
      
          autonomia = 92/40
          autonomia = 2.3 horas




# 1.5 Análise assistida por IA

## 1. Classificação dos dados

###  Variáveis contínuas (numéricas)

*   Temperatura interna (°C)
*   Temperatura externa (°C)
*   Nível de energia (%)
*   Pressão do tanque (bar)

**Características:**

*   Séries temporais
*   Tendência clara (não estacionárias)
*   Forte correlação entre variáveis (energia, pressão e temperatura)

###  Variáveis binárias

*   Integridade estrutural (0/1)

**Características:**

*   Indicador de estado crítico
*   Mudança abrupta (evento de falha)

###  Variáveis categóricas (estado)

**Módulos:**

*   Vida
*   Propulsão
*   Comunicação
*   Térmico

**Valores:**

*   OK
*   WARN
*   FAIL

**Características:**

*   Sistema de estados discretos
*   Evolução sequencial (OK → WARN → FAIL)

---

## 2. Identificação de possíveis anomalias

### Anomalia térmica

*   Temperatura interna sobe continuamente (22 → 48°C)
*   Enquanto externa cai (-55 → -85°C)

**Problema:**

*   Comportamento inverso ao esperado → falha de controle térmico

### Anomalia energética

*   Queda acelerada e constante (25%/h)

**Problema:**

*   Consumo excessivo não justificado apenas por operação normal
*   Provável sobrecarga de sistemas tentando compensar falhas

### Anomalia de pressão

*   Queda contínua (5.2 → 2.9 bar)

**Problema:**

*   Indício de:
    *   vazamento
    *   falha de vedação
    *   consumo anormal

### Anomalia nos módulos (falha em cascata)

**Sequência observada:**

*   Térmico → WARN → FAIL
*   Comunicação → degrada
*   Propulsão → falha
*   Vida → falha final

**Problema:**

*   Forte dependência entre sistemas
*   Falha inicial propagando para todo o sistema

### Anomalia estrutural

*   Integridade muda de 1 → 0 abruptamente

**Problema:**

*   Indica evento crítico (colapso)
*   Provavelmente consequência acumulada:
    *   pressão baixa
    *   estresse térmico

---

## 3. Sugestões de risco

### Risco alto (crítico)

**1. Falha do sistema térmico**

*   **Impacto:** global
*   **Consequência:**
    *   superaquecimento interno
    *   degradação de equipamentos

**Ação sugerida:**

*   Redundância térmica
*   Monitoramento preditivo com alerta precoce

**2. Perda de pressão**

*   **Impacto:** estrutural + operacional

**Ação sugerida:**

*   Sensores redundantes
*   Sistema automático de isolamento de vazamento

**3. Colapso energético**

*   **Impacto:** todos os sistemas

**Ação sugerida:**

*   Controle de carga inteligente (load shedding)
*   Priorização de módulos críticos (vida > comunicação > outros)

### Risco médio

**4. Falha em cascata de módulos**

*   **Problema:** sistêmico (dependência excessiva)

**Ação sugerida:**

*   Arquitetura desacoplada
*   Fail-safe independente por módulo

**5. Falha estrutural tardia**

*   Ocorre após degradação prolongada

**Ação sugerida:**

*   Modelos preditivos baseados em:
    *   temperatura
    *   pressão
    *   vibração (se disponível)

### Risco operacional

**6. Falta de resposta adaptativa**

*   Sistema não reage às falhas iniciais

**Ação sugerida:**

*   Implementar:
    *   controle autônomo
    *   IA de detecção de anomalias em tempo real

---

## Conclusão geral

O sistema apresenta:

*   Degradação contínua (energia e pressão)
*   Falha primária térmica
*   Falha em cascata de módulos
*   Evento final: colapso estrutural

**Classificação global do cenário:**

*   Falha crítica sistêmica com propagação em cascata